backend-api/src/functions/services/clientCredentialsService/clientCredentialsService.ts

```diff
@@ -1,8 +1,71 @@
 import { createHash } from "crypto";
 import { Result, errorResult, successResult } from "../../utils/result";
-import { IRequestBody } from "../../asyncCredential/asyncCredentialHandler";
+import {
+  GetParameterCommand,
+  GetParameterRequest,
+  SSMClient,
+} from "@aws-sdk/client-ssm";
+import { ssmClient } from "./ssmClient";
+
+let cache: CacheEntry | null = null;
+
+export class ClientCredentialsService
+  implements
+    IGetClientCredentials,
+    IValidateTokenRequest,
+    IValidateAsyncCredentialRequest,
+    IGetClientCredentialsById
+{
+  ssmClient: SSMClient;
+  cacheTTL: number;
+
+  constructor() {
+    this.ssmClient = new SSMClient(ssmClient);
+    this.cacheTTL = 3600 * 1000;
+  }
+
+  getClientCredentials = async (): Promise<Result<IClientCredentials[]>> => {
+    if (cache && cache.expiry > Date.now()) {
+      return successResult(cache.data);
+    }
+
+    const command: GetParameterRequest = {
+      Name: "/dev/async-credential/CLIENT_CREDENTIALS",
+      WithDecryption: true, // Parameter is encrypted at rest
+    };
+
+    let response;
+    try {
+      response = await this.ssmClient.send(new GetParameterCommand(command));
+    } catch (e: unknown) {
+      return errorResult("Client Credentials not found");
+    }
+
+    const clientCredentialResponse = response.Parameter?.Value;
+
+    if (!clientCredentialResponse) {
+      return errorResult("Client Credentials is null or undefined");
+    }
+
+    let parsedCredentials;
+    try {
+      parsedCredentials = JSON.parse(clientCredentialResponse);
+    } catch (e: unknown) {
+      return errorResult("Client Credentials is not valid JSON");
+    }
+
+    if (!this.isCredentialsArrayValid(parsedCredentials)) {
+      return errorResult("Parsed Client Credentials array is malformed");
+    }
+
+    cache = {
+      expiry: Date.now() + this.cacheTTL,
+      data: parsedCredentials,
+    };
+
+    return successResult(parsedCredentials);
+  };
 
-export class ClientCredentialsService implements IClientCredentialsService {
   validateTokenRequest = (
```

Comment:
Should we rename this to `validateAsyncTokenRequest` to align with `validateAsyncCredentialRequest`

Reply:
Done